src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -724,7 +912,7 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
   }
 
   private async listDevicesOfKind(deviceKind: string): Promise<MediaDeviceInfo[]> {
-    if (this.deviceInfoCache === null) {
+    if (this.deviceInfoCache === null || !this.isWatchingForDeviceChanges()) {
```

Comment:
Is this really necessary? In what cases can this happen?

Reply:
`listDevicesOfKind` is called by `listAudioDevices`, which is a public API that can be called at any time — including when there are no active devices and no device change observers. For example: immediately after creating the controller!

```typescript
const d = new DefaultDeviceController(…);

// `d` has no global `devicechange` observer so that it doesn't leak.

let audio = await d.listAudioDevices();
…
// We still have no observer, so this needs to update the device info cache again.
audio = await d.listAudioDevices();
```

If we have no device change observers and no active devices, this thing is 'dormant', and has no observers and thus won't leak if discarded… but in this case we need to update our device cache every time, just in case something changed since the last call. If we have observers our device info cache is kept up to date, and we can skip the work.